carbon-p2-plugin/src/main/java/org/wso2/maven/p2/utils/FileManagementUtil.java

```diff
@@ -0,0 +1,286 @@
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+package org.wso2.maven.p2.utils;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Properties;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+import java.util.zip.ZipOutputStream;
+
+public class FileManagementUtil {
+    private static final int BUFFER = 2048;
+
+
+    /**
+     * @param destination path pointing the [carbon product]/repository/components folder
+     * @param profile     name of the profile
+     * @return the config.ini file for the Profile
+     */
+    public static File getProfileConfigIniFile(String destination, String profile) {
+        return new File(destination + File.separator + profile + File.separator + "configuration" +
+                File.separator + "config.ini");
+    }
+
+    /**
+     * Updated the given config.ini file with a given key value pair.
+     *
+     * @param configIniFile File object representing the config.ini
+     * @param propKey       property key
+     * @param value         property value
+     */
+    public static void changeConfigIniProperty(File configIniFile, String propKey, String value) {
+        Properties prop = new Properties();
+
+        try (InputStream inputStream = new FileInputStream(configIniFile)){
+            prop.load(inputStream);
+            prop.setProperty(propKey, value);
+            try(OutputStream outputStream = new FileOutputStream(configIniFile)) {
+                prop.store(outputStream, null);
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Zip a give folder to a give output zip file.
+     *
+     * @param srcFolder   source folder
+     * @param destZipFile path to the output zip file
+     */
+    public static void zipFolder(String srcFolder, String destZipFile) {
+        try(FileOutputStream fileWriter = new FileOutputStream(destZipFile);
+            ZipOutputStream zip = new ZipOutputStream(fileWriter)) {
+            addFolderContentsToZip(srcFolder, zip);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+
+    }
+
+    private static void addToZip(String path, String srcFile, ZipOutputStream zip) {
+        File folder = new File(srcFile);
+        if (folder.isDirectory()) {
+            addFolderToZip(path, srcFile, zip);
+        } else {
+            // Transfer bytes from in to out
+            byte[] buf = new byte[1024];
+            int len;
+            try (FileInputStream inputStream = new FileInputStream(srcFile)){
```

Comment:
This is the migrated code. It is refactored in https://github.com/wso2/carbon-maven-plugins/pull/8